src/pymovements/gaze/io.py

```diff
@@ -360,6 +362,98 @@ def from_asc(
             for fileinfo_key, fileinfo_dtype in column_schema_overrides.items()
         ])
 
+    if experiment is None:
+        experiment = Experiment(sampling_rate=metadata['sampling_rate'])
+    if experiment.screen is None:
+        experiment.screen = Screen()
+    if experiment.eyetracker is None:
+        experiment.eyetracker = EyeTracker()
+
+    # Resolution comparison with ASCII file and experiment
+
+    if experiment.screen.width_px is None or experiment.screen.height_px is None:
+        experiment.screen.width_px, experiment.screen.height_px = metadata['resolution']
+
+    if metadata['resolution'] != (experiment.screen.width_px, experiment.screen.height_px):
+        raise ValueError(
```

Comment:
Great that you check for consistency here.

I think it's ok to be pretty restrictive in this PR, but maybe it's a good idea to raise a warning instead in the future.

As there's quite a lot copy-paste in the following lines it would be nice if you could create a simple private helper function like this one:

```python
def _raise_inconsistent_values(value1: Any, value2: Any, name: str) -> None:
```
The error string can be then parametrized by the input arguments.